homeassistant/components/inels/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""The iNELS integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.devices import Device
+from inelsmqtt.discovery import InelsDiscovery
+from paho.mqtt import MQTTException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+
+from .const import LOGGER
+
+type InelsConfigEntry = ConfigEntry[InelsData]
+
+PLATFORMS: list[Platform] = [
+    Platform.SWITCH,
+]
+
+
+@dataclass
+class InelsData:
+    """Represents the data structure for INELS runtime data."""
+
+    mqtt: InelsMqtt
+    devices: list[Device]
+
+
+def inels_discovery(mqtt: InelsMqtt) -> list[Device]:
+    """Discover iNELS devices."""
+
+    try:
+        i_disc = InelsDiscovery(mqtt)
+        devices: list[Device] = i_disc.discovery()
+    except (MQTTException, ValueError, TimeoutError) as exc:
+        LOGGER.error("Discovery error %s, reason %s", exc.__class__.__name__, exc)
+        mqtt.close()
+        raise ConfigEntryNotReady from exc
+    else:
+        return devices
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: InelsConfigEntry) -> bool:
+    """Set up iNELS from a config entry."""
+
+    mqtt = InelsMqtt(entry.data)
+
+    def connect_and_discover_devices() -> list[Device]:
+        """Test connection and discover devices."""
+        conn_result = mqtt.test_connection()
+        if isinstance(conn_result, int):  # None -> no error, int -> error code
```

Comment:
Can't we just exclude `None`?
```suggestion
        if conn_result is not None:
```